rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -157,10 +158,15 @@ void Player::prepare_publishers()
 {
   auto topics = reader_->get_all_topics_and_types();
   for (const auto & topic : topics) {
+    Rosbag2QoS topic_qos{};
```

Comment:
can you move this logic out into a `qos_for_publisher(topic_name)` function like in the recorder? I'll have similar adaptive logic to add to it